Content.Client/Lobby/UI/HumanoidProfileEditor.xaml.cs

```diff
@@ -649,7 +650,18 @@ public void RefreshAntags()
                 selector.Select(Profile?.AntagPreferences.Contains(antag.ID) == true ? 0 : 1);
 
                 var requirements = _entManager.System<SharedRoleSystem>().GetAntagRequirement(antag);
-                if (!_requirements.CheckRoleRequirements(requirements, (HumanoidCharacterProfile?)_preferencesManager.Preferences?.SelectedCharacter, out var reason))
+                var antagAllSelection = Loc.GetString("ban-panel-role-selection-antag-all-option");
+
+                if (_requirements.IsAntagBanned(new[] { $"Antag:{antag.ID}", $"Antag:{antagAllSelection}" }, out var banReason, out var expirationTime))
+                {
+                    if (banReason != null)
```

Comment:
This API is kind of ass, is there a way to make this better by just letting people pass in the ID directly instead.